build.assets/Dockerfile-ct-ng

```diff
@@ -0,0 +1,57 @@
+# Base on https://github.com/crosstool-ng/crosstool-ng/blob/8825cfc2abf696395bd27bd0e7cea3653004280b/testing/docker/ubuntu22.04/Dockerfile
+FROM ubuntu:22.04 as ct-ng
+
+ARG CTNG_UID=1000
+ARG CTNG_GID=1000
+
+RUN groupadd -g $CTNG_GID ctng
+RUN useradd -d /home/ctng -m -g $CTNG_GID -u $CTNG_UID -s /bin/bash ctng
+
+# Non-interactive configuration of tzdata
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+RUN { echo 'tzdata tzdata/Areas select Etc'; echo 'tzdata tzdata/Zones/Etc select UTC'; } | debconf-set-selections
+
+RUN apt-get update
+RUN apt-get install -y gcc g++ gperf bison flex texinfo help2man make libncurses5-dev \
+    python3-dev autoconf automake libtool libtool-bin gawk wget bzip2 xz-utils unzip \
+    patch libstdc++6 rsync git meson ninja-build
+
+USER ctng
+WORKDIR /home/ctng
+
+RUN wget https://github.com/crosstool-ng/crosstool-ng/releases/download/crosstool-ng-1.26.0/crosstool-ng-1.26.0.tar.bz2 && \
+    tar xf crosstool-ng-1.26.0.tar.bz2 && \
+    cd crosstool-ng-1.26.0 && \
+    ./bootstrap && \
+    ./configure --prefix=/home/ctng/.local/ct-ng && \
+    make -j$(nproc) && \
+    make install && \
+    cd .. && \
+    rm -rf crosstool-ng-1.26.0 crosstool-ng-1.26.0.tar.bz2
+
+ENV PATH="/home/ctng/.local/ct-ng/bin:$PATH"
+
+RUN mkdir -p /home/ctng/build/amd64 && \
+    mkdir -p /home/ctng/build/i686 && \
+    mkdir -p /home/ctng/build/arm64 && \
+    mkdir -p /home/ctng/build/arm && \
+    mkdir -p /home/ctng/src
+COPY ./ct-ng-configs/amd64.config /home/ctng/build/amd64/.config
+COPY ./ct-ng-configs/i686.config /home/ctng/build/i686/.config
+COPY ./ct-ng-configs/arm64.config /home/ctng/build/arm64/.config
+COPY ./ct-ng-configs/arm.config /home/ctng/build/arm/.config
+
+WORKDIR /home/ctng/build
+
+#
+#FROM ubuntu:22.04
+#
+## Create a non-root user with id 1000 - ubuntu
+#RUN useradd -m -u 1000 ubuntu
+#USER ubuntu
+#WORKDIR /home/ubuntu
+#
+#COPY --from=ct-ng /home/ctng/x-tools /home/ubuntu/x-tools
```

Comment:
This looks like a nice 2-phase build.  I'm curious as to why you've commented it out?

Reply:
Forgot to remove. The idea is simple:
1. We build a simple container with `ct-ng`
2. We use that container to build the cross-compiler
3. We build container #2 with minimal dependencies and use it to build Teleport.

For some reason, when I tried to build the cross-compiler as a part of `docker build,` I was getting a weird build failure that I was not able to reproduce anywhere else. This part included the cross-compiler in another stage, but because it was not working, I commented it out and switched to building cross-compiler using `docker run` instead of `docker build`.